e2e/cypress/integration/enterprise/system_console/channel_moderation_spec.js

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+const waitUntilConfigSave = () => {
+    cy.waitUntil(() => cy.get('#saveSetting').then((el) => {
+        return el[0].innerText === 'Save';
+    }));
+};
+
+describe('Channel Moderation Test', () => {
+    before(() => {
+        // Reset permissions in system scheme to defaults.
+        cy.apiLogin('sysadmin');
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+        cy.findByTestId('resetPermissionsToDefault').click();
+        cy.get('#confirmModalButton').click();
+        cy.get('#saveSetting').click();
+        waitUntilConfigSave();
+    });
+
+    it('MM-21789 - Add a group and change the role and then save and ensure the role was updated on channel configuration page', () => {
+        const channelName = 'autem';
+
+        // # Go to system admin page and to channel configuration page of channel "autem"
+        cy.visit('/admin_console/user_management/channels');
+
+        // # Search for the channel.
+        cy.findByTestId('search-input').type(`${channelName}{enter}`);
+        cy.findByTestId(`${channelName}edit`).click();
+
+        // # Wait until the groups retrieved and show up
+        cy.wait(500); //eslint-disable-line cypress/no-unnecessary-waiting
+
+        // # Uncheck all the boxes currently checked (align with the system scheme permissions)
+        cy.findByTestId('create_post-guests').click();
+        cy.findByTestId('create_post-members').click();
+        cy.findByTestId('create_reactions-members').click();
```

Comment:
The current logic to uncheck all the checkboxes is not correct. Since the test works on a single channel every time, if the test had failed previously in an abrupt way, i.e. some checkboxes checked and some not checked, then the next time we run the test, it will fail. The correct way of unchecking the checkbox is to check if the element has a class "checked" and only then uncheck the same.